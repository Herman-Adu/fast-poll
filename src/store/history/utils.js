export const sortPollsByProperty = (polls, prop, order = 'desc') => {
  if (order === 'desc') {
    return polls.sort((a, b) => {
      if (prop === 'date') {
        return Date.parse(b[prop]) - Date.parse(a[prop]);
      }

      return b[prop] - a[prop];
    });
  } else if (order === 'asc') {
    return polls.sort((a, b) => {
      if (prop === 'date') {
        return Date.parse(a[prop]) - Date.parse(b[prop]);
      }

      return a[prop] - b[prop];
    });
  }
};
