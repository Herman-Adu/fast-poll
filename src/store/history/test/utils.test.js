import * as utils from '../utils'

describe('history utils', () => {
  const polls = [{
    date: "2018-06-25T23:03:45.174Z",
    id: '124',
    title: 'created poll 3',
    order: 3,
  }, {
    date: "2018-04-25T23:03:45.174Z",
    id: '1',
    title: 'created poll 1',
    order: 1,
  }, {
    date: "2018-05-25T23:03:45.174Z",
    id: '12',
    title: 'created poll 2',
    order: 1,
  }]

  it('sortPollsByProperty sorts polls given a property in acsending order', () => {
    expect(utils.sortPollsByProperty(polls, 'order')).toMatchSnapshot()
  })

  it('sortPollsByProperty sorts polls given a property in descending order', () => {
    expect(utils.sortPollsByProperty(polls, 'order', 'desc')).toMatchSnapshot()
  })

  it('sortPollsByProperty sorts polls by date', () => {
    expect(utils.sortPollsByProperty(polls, 'date')).toMatchSnapshot()
  })
})