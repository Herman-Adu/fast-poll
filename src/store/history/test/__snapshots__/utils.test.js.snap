// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`history utils sortPollsByProperty sorts polls by date 1`] = `
Array [
  Object {
    "date": "2018-04-25T23:03:45.174Z",
    "id": "1",
    "order": 1,
    "title": "created poll 1",
  },
  Object {
    "date": "2018-05-25T23:03:45.174Z",
    "id": "12",
    "order": 1,
    "title": "created poll 2",
  },
  Object {
    "date": "2018-06-25T23:03:45.174Z",
    "id": "124",
    "order": 3,
    "title": "created poll 3",
  },
]
`;

exports[`history utils sortPollsByProperty sorts polls given a property in acsending order 1`] = `
Array [
  Object {
    "date": "2018-04-25T23:03:45.174Z",
    "id": "1",
    "order": 1,
    "title": "created poll 1",
  },
  Object {
    "date": "2018-05-25T23:03:45.174Z",
    "id": "12",
    "order": 1,
    "title": "created poll 2",
  },
  Object {
    "date": "2018-06-25T23:03:45.174Z",
    "id": "124",
    "order": 3,
    "title": "created poll 3",
  },
]
`;

exports[`history utils sortPollsByProperty sorts polls given a property in descending order 1`] = `
Array [
  Object {
    "date": "2018-06-25T23:03:45.174Z",
    "id": "124",
    "order": 3,
    "title": "created poll 3",
  },
  Object {
    "date": "2018-04-25T23:03:45.174Z",
    "id": "1",
    "order": 1,
    "title": "created poll 1",
  },
  Object {
    "date": "2018-05-25T23:03:45.174Z",
    "id": "12",
    "order": 1,
    "title": "created poll 2",
  },
]
`;
